### DISCUSSION TITLE
Generating Attention Weights From GATCONV

### DISCUSSION THREAD

**Discussion by akul-goyal** [pyg-team/pytorch_geometric#4036] [Q&A] [answered]
Hi, I am trying to understand the embedding of certain nodes within my graph. I am using a 2 layers Graph Attention Neural Network to create the embeddings. I want to find out the weights that are placed for each of the nodes within a k-hops of a target node. Is there a way for me to do this?

**Comment by rusty1s** [comment, marked as answer]
You can get the attention-based adjacency matrix from `GATConv` by making use of `return_attention_weights=True` in its `forward` call. Together with `utils.k_hop_subgraph()`, you should be able to gather the weights of a node within its k-hop neighborhood.

**Reply by akul-goyal** [reply]
Thanks for the quick response. I will give it a try!
